dask/array/core.py

```diff
@@ -2660,6 +2660,10 @@ def bincount(x, weights=None, minlength=None):
 
     return Array(dsk, name, chunks, dtype)
 
+@wraps(np.digitize)
+def digitize(a, bins, right=False):
+    func = partial(np.digitize, bins=bins, right=right)
+    return elemwise(func, a)
```

Comment:
Partition is intended for use with dask.dataframe and dask.bag, which are linear containers.  Block is more often used in dask.array, which has an n-dimensional structure.  The term "partition" has historically be used with linear containers and the term "block" or "chunk" has historically been used with nd-arrays.  Hence the difference.